src/status_im/ui/screens/wallet/send/views/amount.cljs

```diff
@@ -0,0 +1,314 @@
+     [react/view {:style (assoc components.styles/flex :background-color :white)}
+      [list/flat-list {:default-separator? false ;true
+                       :data               assets
+                       :key-fn             (comp str :symbol)
+                       :render-fn          #(do
+                                              [react/touchable-highlight {:on-press       (fn [] (on-asset %))
+                                                                          :underlay-color colors/black-transparent}
+                                               (render-token-item %)])}]]]))
+
+(defn show-current-asset [{:keys [name icon decimals amount] :as token}]
+  [react/view {:style {:flex-direction     :row,
+                       :justify-content    :center
+                       :padding-horizontal 21
+                       :padding-vertical   12}}
+   [list/item-image icon]
+   [react/view {:margin-horizontal 9
+                :flex              1}
+    [list/item-content
+     [react/text {:style {:margin-right 10,
+                          :font-weight  "500"
+                          :font-size    15
+                          :color        colors/white}} name]
+     [react/text {:style           {:font-size   14
+                                    :padding-top 4
+                                    :color       colors/white-transparent}
+                  :ellipsize-mode  :middle
+                  :number-of-lines 1}
+      (str (wallet.utils/format-amount amount decimals)
+           " "
+           (wallet.utils/display-symbol token))]]]
+   list/item-icon-forward])
+
+(defn fetch-token [all-tokens network token-symbol]
+  {:pre [(map? all-tokens) (map? network)]}
+  (when (keyword? token-symbol)
+    (tokens/asset-for all-tokens
+                      (ethereum/network->chain-keyword network)
+                      token-symbol)))
+
+(defn create-initial-state [{:keys [symbol decimals]} amount]
+  {:input-amount  (when amount
+                    (when-let [amount' (money/internal->formatted amount symbol decimals)]
+                      (str amount')))
+   :inverted      false
+   :edit-gas      false
+   :error-message nil})
+
+(defn input-currency-symbol [{:keys [inverted] :as state} {:keys [symbol] :as coin} {:keys [code]}]
+  {:pre [(boolean? inverted) (keyword? symbol) (string? code)]}
+  (if-not (:inverted state) (wallet.utils/display-symbol coin) code))
+
+(defn converted-currency-symbol [{:keys [inverted] :as state} {:keys [symbol] :as coin} {:keys [code]}]
+  {:pre [(boolean? inverted) (keyword? symbol) (string? code)]}
+  (if (:inverted state) (wallet.utils/display-symbol coin) code))
+
+(defn valid-input-amount? [input-amount]
+  (and (not (string/blank? input-amount))
+       (not (:error (wallet.db/parse-amount input-amount 100)))))
+
+(defn converted-currency-amount [{:keys [input-amount inverted]} token fiat-currency prices]
+  (when (valid-input-amount? input-amount)
+    (if-not inverted
```

Comment:
it could be
```clj
(some-> (common/token->fiat-conversion prices token fiat-currency input-amount)
        (money/with-precision (if inverted 8 2)))
```